Sources/SKSupport/Collection.swift

```diff
@@ -10,7 +10,23 @@
 //
 //===----------------------------------------------------------------------===//
 
-extension RandomAccessCollection where Element: Equatable {
+extension RandomAccessCollection where Element: Equatable & Hashable {
+
+  /// Searches back through both patterns until we find an element that
+  /// doesn’t match or until we’ve reached the beginning of the pattern.
+  @inlinable
+  public func backwards<Pattern>(on pattern: Pattern, i: Index) -> Index? where Pattern: RandomAccessCollection, Pattern.Element == Element {
```

Comment:
How does this compare to something like `self[i...].starts(with: pattern)` ?  This whole helper function is pretty weird as an API and there's no real point in returning the index since it's always a known offset from `i`.  If we need to keep a helper like this it should be private at least.

Reply:
Err, I messed up that suggestion, but something along the lines of `self[index(i, offsetBy: -pattern.count)..<i] == pattern`.  Basically, we don't care whether this walks backwards of forwards (right?) and the return value is equivalent to a Bool, since we know the result is always either `nil` or `index(i, offsetBy: -pattern.count)`, although I might have an off-by-one error.

Reply:
I think this algorithm aims to compare the pattern to the text from right to left instead of left to right.

Reply:
^^ Just making sure you saw these comments, @abdullahselek.

Reply:
@abdullahselek but why does it matter?  It always returns either `nil` or `index(i, offsetBy: -pattern.count)`.   It feels like this code is overly complicated for what it's doing.  If searching backwards has an advantage, it would be nice to better understand what that is, and even still maybe we don't need to return `Index`.